presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -446,11 +446,13 @@ public static Optional<ConnectorPageSource> createHivePageSource(
                 tableToPartitionMapping,
                 fileSplit,
                 tableBucketNumber);
-
+// at this point synthetics should have a prefilled value
         Set<Integer> outputIndices = hiveColumns.stream()
                 .map(HiveColumnHandle::getHiveColumnIndex)
                 .collect(toImmutableSet());
 
+// this next line removes the synthetic columns. The problem is that row_id has columnMappingKind = regular
```

Comment:
this comment and the one above are not very helpful and also look incorrect (currently it would get prefilled as the column mapping kind. with your commented out code it. would get postfilled).

Reply:
rewritten